android/src/main/java/com/johnsonsu/rnsoundplayer/RNSoundPlayerModule.java

```diff
@@ -125,11 +145,32 @@ public void onCompletion(MediaPlayer arg0) {
           }
       });
     } else {
-      Uri uri = Uri.parse("android.resource://" + getReactApplicationContext().getPackageName() + "/raw/" + name);
+      Uri uri;
+
+      int soundResID = getReactApplicationContext().getResources().getIdentifier(name, "raw", getReactApplicationContext().getPackageName());
+
+      if (soundResID > 0) {
+        uri = Uri.parse("android.resource://" + getReactApplicationContext().getPackageName() + "/raw/" + name);
+      } else {
+        String folder = getReactApplicationContext().getFilesDir().getAbsolutePath();
```

Comment:
Since the code inside `else` is the same for both cases, I think we should move them into a private function, maybe `private Uri getUriFromFile(String name, String type)`?

Reply:
Good idea. Done.